yarn-project/archiver/src/archiver/archiver_store.ts

```diff
@@ -50,6 +50,15 @@ export interface ArchiverDataStore {
    */
   addBlocks(blocks: L1Published<L2Block>[]): Promise<boolean>;
 
+  /**
+   * Unwinds blocks from the database
+   * @param from -  The tip of the chain, passed for verification purposes,
```

Comment:
The "tip of the chain" in the implementation is the most recently synced L2 block, but the two might be different.

Reply:
Can you clarify a bit. I'm not following. Both of the values are based on the synched L2 block, e.g., the from and the value we are checking against, so we should not be updating it in the middle.

Reply:
Got it- my question/concern was around how does the caller know what to pass as "from", and how would it know that it is the tip.

Reply:
The caller should only really be the archiver, and it will be able to just fetch the information from the store directly. It so mainly to avoid someone doing a call to it directly.